metadata-io/src/test/java/com/linkedin/metadata/entity/EntityServiceTest.java

```diff
@@ -1190,6 +1192,55 @@ public void testRestoreIndices() throws Exception {
         }
     }
 
+    @Test
+    public void testValidateUrn() throws Exception {
+        // Valid URN
+        Urn validTestUrn = new Urn("li" , "testType", new TupleKey("testKey"));
+        EntityService.validateUrn(validTestUrn);
+
+        // URN with trailing whitespace
+        Urn testUrnWithTrailingWhitespace = new Urn("li" , "testType", new TupleKey("testKey   "));
+        try {
+            EntityService.validateUrn(testUrnWithTrailingWhitespace);
+            Assert.fail("Should have raised IllegalArgumentException for URN with trailing whitespace");
+        } catch (IllegalArgumentException e){
+            assertEquals(e.getMessage(), "Error: cannot provide an URN with leading or trailing whitespace");
+
+        }
+
+        // Urn purely too long
+        StringBuilder buildStringTooLong = new StringBuilder();
+        for (int i = 0; i < 510; i++) {
+            buildStringTooLong.append('a');
+        }
+
+        Urn testUrnTooLong = new Urn("li", "testType", new TupleKey(buildStringTooLong.toString()));
+        try {
+            EntityService.validateUrn(testUrnTooLong);
+            Assert.fail("Should have raised IllegalArgumentException for URN too long");
+        } catch (IllegalArgumentException e){
+            assertEquals(e.getMessage(), "Error: cannot provide an URN longer than 512 bytes (when URL encoded)");
+        }
+
+        // Urn too long when URL encoded
+        StringBuilder buildStringTooLongWhenEncoded = new StringBuilder();
+        StringBuilder buildStringSameLengthWhenEncoded = new StringBuilder();
+        for (int i = 0; i < 200; i++){
```

Comment:
Checkstyle's gonna complain on this I think
```suggestion
        for (int i = 0; i < 200; i++) {
```